exchanges/websocket/wshandler/wshandler.go

```diff
@@ -620,14 +620,14 @@ func (w *Websocket) CanUseAuthenticatedEndpoints() bool {
 	return w.canUseAuthenticatedEndpoints
 }
 
-// AddResponseWithID adds data to IDResponses with locks and a nil check
-func (w *WebsocketConnection) AddResponseWithID(id int64, data []byte) {
+// SetResponseIDAndData adds data to IDResponses with locks and a nil check
+func (w *WebsocketConnection) SetResponseIDAndData(id int64, data []byte) {
 	w.Lock()
-	defer w.Unlock()
```

Comment:
Achieves the same result, it's easier to know your lock will get unlocked if you put a defer straight after, otherwise you run into scenarios like this: https://github.com/thrasher-corp/gocryptotrader/pull/446/files#diff-c756b319860bc00c5b63bb5399dbb40bR554